api/v1alpha1/progressiverollout_types.go

```diff
@@ -73,13 +73,15 @@ type ProgressiveRolloutStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	Stages     []StageStatus      `json:"stages,omitempty"`
```

Comment:
we use pointer semantics practically everywhere in this; wondering if it would be worth it to make it a list of pointers.
```suggestion
	Stages     []*StageStatus      `json:"stages,omitempty"`
```

Reply:
The reasons why I suggested value semantics are:
- we use value semantics everywhere, and I wanted to keep consistency
- for `metav1.Condition`, when I looked at some operators like https://github.com/fluxcd/kustomize-controller/blob/main/api/v1beta1/kustomization_types.go#L212 or https://github.com/fluxcd/flagger/blob/main/pkg/apis/flagger/v1beta1/status.go#L85 or https://github.com/argoproj/argo-cd/blob/master/pkg/apis/application/v1alpha1/types.go#L544 they all seems to use value semantic
- there is an official package to deals with Conditions at https://pkg.go.dev/k8s.io/apimachinery/pkg/api/meta but those function requires a pointer to the slice. This is why we have the `GetStatusConditions()` method at https://github.com/Skyscanner/argocd-progressive-rollout/blob/main/api/v1alpha1/progressiverollout_types.go#L79

So to me it makes sense to keep the same semantic, and possibly the same names for our functions, to interact with StageStatus. This is why we have a `GetStageStatus()` method for example.

I'm happy to discuss if we're not doing a great job at the using the same approach as the `api/meta` package, or if we would be better with a different approach, but would feel weird to have values for every slice _except_ for this one

Reply:
this is a good rationale;
let's make sure we're not getting confused with this and that the "true" value resides with this one.